File: backend/app/schemas/sponsor.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List

class SponsorCreate(BaseModel):
    event_id: int
    name: str
    contact_email: Optional[str] = None
    amount: float
    sponsorship_type: Optional[str] = "Cash"

class SponsorResponse(BaseModel):
    id: int
    event_id: int
    name: str
    contact_email: Optional[str] = None
    amount: float
    sponsorship_type: Optional[str] = None
    received_at: datetime

    class Config:
        from_attributes = True

class FinancialSummary(BaseModel):
    event_id: int
    total_budget: float
    total_sponsorship: float
    total_expenses: float
    net_balance: float
    sponsors: List[SponsorResponse]

File: frontend/src/types/sponsor.ts
export interface Sponsor {
  id: number;
  event_id: number;
  name: string;
  contact_email?: string;
  amount: number;
  sponsorship_type?: string;
  received_at: string;
}

export interface SponsorCreate {
  event_id: number;
  name: string;
  contact_email?: string;
  amount: number;
  sponsorship_type?: string;
}

export interface FinancialSummary {
  event_id: number;
  total_budget: number;
  total_sponsorship: number;
  total_expenses: number;
  net_balance: number;
  sponsors: Sponsor[];
}